### PR TITLE
ToDo list add remove test

### PR DESCRIPTION
**Pull request by ogagaoghene**
# ToDo list add-remove test

## Description
Add and remove test functionalities are implemented

### CODE REVIEW

File: src/test/add-remove.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Tasks.addTask(task);
+    displayTasks(Tasks.tasksArray, listContainer);
+    expect(Tasks.tasksArray).toHaveLength(1);
+  });
+
+  test('Test local storage', () => {
+    JSON.parse(localStorage.getItem('todos'));
+    expect(localStorage).toHaveLength(1);
+  });
+});
```
Comment (BertrandConxy) [inline src/test/add-remove.test.js]: - [ ] It looks like your tests are not passing. There is a note saying that `local storage` is not defined. I would advise you to use mocks that imitate the functioning of the local storage and by doing this, you will be able to make your tests pass. Here is a [document](https://bholmes.dev/blog/mocking-browser-apis-fetch-localstorage-dates-the-easy-way-with-jest/) to help you learn more about mocks. Also this [post](https://stackoverflow.com/questions/32911630/how-do-i-deal-with-localstorage-in-jest-tests) on stack overflow. 😺 
image

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @ogagaoghene ✋, 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💪 
- no linter errors
- good PR description

## Required Changes ♻️

- [x] It was difficult for me to find the way to run the tests because you have not stated the procedures to Run the tests. You can put the procedures in your README file or just here in the PR description. I tried to use the usual way which running is `npm test` but it was not working because you have not defined it correctly. Take a look at your scripts in the package.json file 👇 
image
Please change it to `test": "jest"` because you are using jest to write and run the tests 😃 💡. By doing this, everyone will be able to run the tests using the command `npm test`

_Check other comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me  @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/add-remove.test.js (hunk 1 above)
Comment (ogagaoghene) [reply src/test/add-remove.test.js]: Thanks for your professional review. I have mocked local storage by using jsdom. All tests have passed. 
image

**Comment by ogagaoghene** [comment]
I appreciate your professional review and I look forward to having positive feedback so I can proceed with the other activities. Thanks in anticipation.

**Review by dasileker — APPROVED** [review]
### **Hi @ogagaoghene,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ogagaoghene** [comment]
Many thanks to you

**Pull request merged**